src/toil/jobStores/aws/jobStore.py

```diff
@@ -74,38 +82,70 @@ def copyKeyMultipart(srcKey, dstBucketName, dstKeyName, headers=None):
     :rtype: boto.s3.multipart.CompletedMultiPartUpload
     :return: An object representing the completed upload.
     """
+    def copyPart(partIndex):
+        if exceptions:
+            return None
+        else:
+            try:
+                for attempt in retry_s3():
+                    with attempt:
+                        # Note that part_num is indexed from 1 while start and end are indexed from
+                        # 0. Notice also that the partIndex variable is indexed from 0.
+                        part = upload.copy_part_from_key(src_bucket_name=srcKey.bucket.name,
+                                                         src_key_name=srcKey.name,
+                                                         src_version_id=srcKey.version_id,
+                                                         part_num=partIndex + 1,
+                                                         start=partIndex * partSize,
+                                                         end=min(partIndex * partSize + partSize,
+                                                                 totalSize) - 1,
+                                                         headers=headers)
+                log.debug('Successfully copied part %d of %d' % (partIndex, totalParts))
+
+            except Exception as e:
+                if partIndex <= 4:
```

Comment:
Not the first four parts of the key. The first four failures. With this code, if the first four parts succeed but another part fails, weird things are going to happen.